canonical/discourse-rad-parser
# fix: Alter dropdown logic to only affect selected dropdown

## Done

- Alters `attachDropdownEvents` to make sure only the selected dropdown and its related content is affected

## QA

- The only way I could test this was to add the `main.js` script manually into `base.html` on juju.is https://github.com/canonical/juju.is/blob/dba04d892aaa2b49ca814223a630afb634d65ef3/templates/docs/base.html#L32. This needs to be wrapped in:
```
<script>
  document.addEventListener('DOMContentLoaded', () => {
    {main.js script here}
    DiscourseRADParser();
  });
</script>
```
^ this will make sure the script runs locally

- Then, run `dotrun` in juju.is repo
  - Go to http://0.0.0.0:8041/docs/juju/tutorial#tear-down-your-test-environment and change the dropdown content here, you should be able to switch back and forth between the options without the page jumping, and the other dropdowns should maintain their state

## Issue / Card

Fixes https://warthogs.atlassian.net/browse/WD-13364
